### ISSUE TITLE
[Feature Request] Revive/rewrite QVM PowerPC

### ISSUE THREAD

**Issue by runlevel5** [ec-/Quake3e#232]
Firstly, I have to say that I love quake3e source port. To me it is one of the BEST Q3 source port. I've been playing ioq3 with friends within the PowerPC community for quite awhile (all thanks to the great support for PPC in the original source code of ioq3). Is there any chance that quake3e could restore the support ppc64 QVM? (specifically restoring the `qcommon/vm_powerpc_*` files)? Many thanks

**Comment by ec-** [comment]
I rewrote QVM several times so 'restore' is incorrect term there - thus, PPC QVM it must be re-written too. It is a time-consuming task so no guarantees from my side

**Comment by KuehnhammerTobias** [comment]
@ec- Just asking out of curiosity, doesn't it work to include the files and compile them?
I own no PPC, and I know this isn't the right place to ask, but maybe adding the files from ioq3 will help others (with a PPC and  some coding skills) to do the time-consuming job for you.
Just some thoughts.

**Comment by runlevel5** [comment]
>  @ec I rewrote QVM several times so 'restore' is incorrect term there - thus, PPC QVM it must be re-written too. It is a time-consuming task so no guarantees from my side

Yes I am aware of this. I am more than happy to offer you access to hardware to work on if you have capacity. 

> @KuehnhammerTobias ..but maybe adding the files from ioq3 will help others (with a PPC and some coding skills) to do the time-consuming job for you.

That's what I thought too but initial attempt did not compile, some functions are reported missing, for example `currentVM`. In theory we could restore this legacy QVM PPC but I need to dig in the git history to work out what's changed

**Comment by KuehnhammerTobias** [comment]
@runlevel5

> That's what I thought too but initial attempt did not compile, some functions are reported missing, for example currentVM. In theory we could restore this legacy QVM PPC but I need to dig in the git history to work out what's changed
That is unfortune, as far as the git history is concerned, ppc_vm files were dropped on [Apr 21, 2017](https://github.com/ec-/Quake3e/commit/f627f04a2ca2dc4bdc5f3c43e308ed8af41af5f9) but I think you found this out already by yourself. Well, assuming the PPC vm work until at that point, it would be the starting point for you to check the code changes.
I'm not sure if that will help you, sorry!

**Comment by runlevel5** [comment]
@ec- So if I could backport the `vm_powerpc.c` from the original ioq3, would you accept such patches given into consideration it is not yet rewritten?

**Comment by ec-** [comment]
@runlevel5 sure it is ok if you can adopt it to new QVM structure and run some mod without errors
